Sources/FirebaseAuth/FirebaseAuthError.swift

```diff
@@ -2,14 +2,8 @@
 
 import Foundation
 
-public struct AuthError {
-}
-
-extension AuthError: Error {
-}
-
-extension AuthError {
-  public enum Code: Int {
+public struct AuthErrorCode: Error {
```

Comment:
I assume that the public API expectation is the `AuthErrorCode` type?  Given that it was previously `AuthError`, this seems odd to me.  If it just is meant be an `Error` type, I wonder if we can flatten this and just add a `public var code: Self { self }` extension on the `FIRAuthErrorCode` enumeration and make it conform to `Error` as a simpler representation.

Reply:
The expected type returned is actually (I think) an `NSError` since it provides a `code` function as well (I assume this is what `NS_ENUM_ERROR` is doing under the hood). Making `FIRAuthErrorCode` conform the Error  could work, I guess I'm just thinking if I should try to produce an actual `NSError` instead  (I don't know if there is more code that we have yet to port which will start to rely on say `domain` or other bits of the NSError api.

Let me flatten this for the time being and see where that gets us.